nowcasting_dataset/data_sources/satellite/satellite_data_source.py

```diff
@@ -86,23 +95,44 @@ def get_spatial_region_of_interest(
         Returns:
             The selected data around the center
         """
-        x_index = (
-            np.searchsorted(data_array.x.values, x_center_osgb) - 1
-        )  # To have the center fall within the pixel
-        y_index = np.searchsorted(data_array.y.values, y_center_osgb) - 1
-        min_y = y_index - (self._square.size_pixels // 2)
-        min_x = x_index - (self._square.size_pixels // 2)
-        assert min_y >= 0, (
-            f"Y location must be at least {(self._square.size_pixels // 2)} "
-            f"pixels from the edge of the area, but is {y_index} for y center of {y_center_osgb}"
-        )
-        assert min_x >= 0, (
-            f"X location must be at least {(self._square.size_pixels // 2)}"
-            f" pixels from the edge of the area, but is {x_index} for x center of {x_center_osgb}"
+        # Get the index into x and y nearest to x_center_osgb and y_center_osgb:
```

Comment:
This new code for `_open_data()` should be functionally equivalent to the old code.  Except with more helpful errors, and hopefully better variable names.